R/derive_vars_crit_flag.R

```diff
@@ -0,0 +1,161 @@
+#'  `"Y"`, `"N"`, and `NA`.
+#' - the `CRITy` variable is populated correctly, i.e.,
+#'   - set to a constant value within a parameter if `CRITyFL` is populated with
+#'   `"Y"`, `"N"`, and `NA` and
+#'   - set to a constant value within a parameter if the criterion condition is
+#'   fulfilled and to `NA` otherwise if `CRITyFL` is populated with `"Y"`,  and
+#'   `NA`
+#'
+#' @param dataset Input dataset
+#' @param crit_nr The criterion number, i.e., the `y` in `CRITy`
+#'
+#'   *Permitted Values*: a positive integer
+#' @param condition Condition for flagging records
+#'
+#'   See description of the `values_yn` argument for details on how the
+#'   `CRITyFL` variable is populated.
+#'
+#'   *Permitted Values*: an unquoted expression which evaluates to a logical (in
+#'    `dataset`)
+#' @param description The description of the criterion
+#'
+#'   The `CRITy` variable is set to the specified value.
+#'
+#'   An expression can be specified to set the value depending on the parameter.
+#'   Please note that the value must be constant within a parameter.
+#'
+#'   *Permitted Values*: an unquoted expression which evaluates to a character
+#'    (in `dataset`)
+#' @param values_yn Should `"Y"` and `"N"` be used for `CRITyFL`?
+#'
+#'   If set to `TRUE`, the `CRITyFL` variable is set to `"Y"` if the condition
+#'   (`condition`) evaluates to `TRUE`, it is set to `"N"` if the condition
+#'   evaluate to `FALSE`, and to `NA` if it evaluates to `NA`.
+#'
+#'   Otherwise, the `CRITyFL` variable is set to `"Y"` if the condition
+#'   (`condition`) evaluates to `TRUE`, and to `NA` otherwise.
+#'
+#'   *Permitted Values*: `TRUE`, `FALSE`
+#' @param create_numeric_flag Create a numeric flag?
+#'
+#'   If set to `TRUE`, the `CRITyFLN` variable is created. It is set to `1` if
+#'   `CRITyFL == "Y"`, it set to `0` if `CRITyFL == "N"`, and to `NA` otherwise.
+#'
+#'   *Permitted Values*: `TRUE`, `FALSE`
+#' @return The input dataset with the variables `CRITy`, `CRITyFL`, and
+#'   optionally `CRITyFLN` added.
+#'
+#' @family der_bds_findings
+#' @keywords der_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' adbds <- tribble(
+#'   ~PARAMCD, ~AVAL,
+#'   "AST",    42,
+#'   "AST",    52,
+#'   "AST",    NA_real_,
+#'   "ALT",    33,
+#'   "ALT",    51
+#' )
+#'
+#' # Create a criterion flag with values "Y" and NA
+#' derive_vars_crit_flag(
+#'   adbds,
+#'   condition = AVAL > 50,
+#'   description = "Absolute value > 50"
+#' )
+#'
+#' # Create criterion flag with values "Y", "N", and NA and parameter dependent
+#' # criterion description
+#' derive_vars_crit_flag(
+#'   adbds,
+#'   crit_nr = 2,
+#'   condition = AVAL > 50,
+#'   description = paste(PARAMCD, "> 50"),
+#'   values_yn = TRUE,
+#'   create_numeric_flag = TRUE
+#' )
+derive_vars_crit_flag <- function(dataset,
+                                  crit_nr = 1,
+                                  condition,
+                                  description,
```

Comment:
Do we use description any where else?  Wondering if there is another argument we could use that is similar to other functions??

Reply:
I don't - just wanted to be consistent.  Thanks for looking into other examples